tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -208,6 +210,178 @@ def setUp(self):
             alert_rule_trigger=self.alert_rule_trigger_critical,
         )
 
+    def mock_migrate_alert_rule(self, metric_alert) -> tuple[
+        DataSource,
+        DataConditionGroup,
+        Workflow,
+        Detector,
+        DetectorState,
+        AlertRuleDetector,
+        AlertRuleWorkflow,
+        DetectorWorkflow,
+    ]:
+        """
+        Create mock data for a dual-written alert rule.
```

Comment:
i wouldn't consider these "mocks" from a testing sense, this is more what i was thinking: https://docs.python.org/3/library/unittest.mock.html -- it allows us to introspect more about the data / calls, and it specifically wouldn't create a model in the db. the reason for doing that is it substantially speeds up tests (there's a bit of a cost in that we're not actually testing with the real models, which has it's own set of issues).